torch/csrc/autograd/engine.cpp

```diff
@@ -590,6 +610,20 @@ auto Engine::evaluate_function(NodeTask& task) -> void {
           continue;
         }
       }
+
```

Comment:
What about the "      // The function already has a buffer"  case below? Why does that not need similar logic?

Reply:
Good catch, I believe it does.

Reply:
Put the logic into input_buffer.